wegodang/users/views.py

```diff
@@ -1,3 +1,54 @@
-from django.shortcuts import render
+import jwt
+import requests
+import json
 
-# Create your views here.
\ No newline at end of file
+from django.http            import JsonResponse
+from django.core.exceptions import ValidationError
+from django.views           import View
+from django.conf            import settings
+
+from users.models           import User
+from users.validator        import validate_email
+
+class KaKaoSignUpView(View):
+    def get(self, request):
+        access_token   = request.GET.get('access_token')
+        KAKAO_INFO_API = 'https://kapi.kakao.com/v2/user/me'
+        user_info      = requests.get(KAKAO_INFO_API, headers={'Authorization': f'Bearer {access_token}'}).json()
+        kakao_id       = user_info['id']
+        name           = user_info['properties']['nickname']
+        email          = user_info['kakao_account']['email']
+
+        result = {
+            'kakao_id' : kakao_id,
+            'username' : name,
+            'email'    : email
+        }
+
+        return JsonResponse({'result' : result }, status = 200)
+    
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            validate_email(data['email'])
+
+            user, is_created = User.objects.get_or_create(
+                kakao_id = data['kakao_id'],
+                defaults = {
+                    'username' : data['username'],
+                    'email'    : data['email']
+                }
+            )
+        
+            status       = 201 if is_created else 200
+            message      = 'FIRST_LOGIN' if is_created else 'LOGIN'
+            client_token = jwt.encode({'id':user.id}, settings.SECRET_KEY, settings.ALGORITHM)
+        
+            return JsonResponse({'message': message, 'token': client_token}, status=status)
+
+        except ValidationError as error:
+            return JsonResponse({"message" : error.message}, status = 400)
+
+        except:
+            return JsonResponse({"message" : "KEYERROR"}, status = 400)
```

Comment:
단순히 except만 사용하면 모든 에러에 대해 예외처리를 진행하기 때문에 꼭 어떤 상황에 대한 예외처리인지를 나타낼 수 있어야 합니다.
```suggestion
        except KeyError:
            return JsonResponse({"message" : "KEYERROR"}, status = 400)
```

Reply:
수정했습니다 !